torch/csrc/jit/passes/onnx/constant_map.cpp

```diff
@@ -0,0 +1,110 @@
+#include <torch/csrc/jit/passes/onnx/constant_map.h>
+
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+// Meyer’s Singleton for C++ 14
+ConstantValueMap& ConstantValueMap::getInstance() {
+  static ConstantValueMap s;
+  return s;
+}
+
+void ConstantValueMap::SetRank(
+    const std::string& tensorName,
+    size_t rankValue) {
+  ConstantValueMap::getInstance().rankMap.emplace(tensorName, rankValue);
+}
+
+bool ConstantValueMap::HasRank(const std::string& tensorName) {
+  return ConstantValueMap::getInstance().rankMap.find(tensorName) !=
+      ConstantValueMap::getInstance().rankMap.end();
+}
+
+c10::optional<size_t> ConstantValueMap::GetRank(const std::string& tensorName) {
+  if (!HasRank(tensorName)) {
+    return c10::nullopt;
+  }
+  return ConstantValueMap::getInstance().rankMap[tensorName];
```

Comment:
nit: add const to this function, and use `.at(tensorName)` which can operate on const unordered_map obj.

Reply:
This is static function, so I could not add const to it.